src/ring_theory/dedekind_domain/selmer_group.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import data.zmod.quotient
+import ring_theory.dedekind_domain.adic_valuation
+import ring_theory.localization.basic
+
+/-!
+# Selmer groups
```

Comment:
This isn't a complete module doc. Please explain what's in the file, implementation details, notation, etc! See the [doc guidelines](https://leanprover-community.github.io/contribute/doc.html).